src/jit/optimizer.cpp

```diff
@@ -3304,7 +3304,7 @@ bool Compiler::optReachWithoutCall(BasicBlock* topBB, BasicBlock* botBB)
 
     // We can always check topBB and botBB for any gc safe points and early out
 
-    if ((topBB->bbFlags | botBB->bbFlags) & BBF_GC_SAFE_POINT)
+    if ((topBB->bbFlags & BBF_GC_SAFE_POINT) || (!endsWithTailCall && (botBB->bbFlags & BBF_GC_SAFE_POINT)))
```

Comment:
If the block has additional normal calls that preceed the final tail call then it is appropriate that the block be flagged with `BBF_GC_SAFE_POINT`